model/datastore.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+const Joi = require('joi');
+const table = Joi.string().required();
+const id = Joi.string().hex().required();
+const idReferenceObject = Joi.object().keys({
+    table,
+    params: Joi.object().keys({
+        id
+    }).required()
+});
+const dataReferenceObject = Joi.object().keys({
+    table,
+    params: Joi.object().keys({
+        id,
+        data: Joi.object().required()
+    }).required()
+});
+const paginateReferenceObject = Joi.object().keys({
+    table,
+    params: Joi.object(),
```

Comment:
do we do anything with params right now? Should we just add this in when we start doing params?